mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -2096,12 +2118,50 @@ populateHostHarnessLogic(ModuleOp &module,
       printp = printResults.getValue();
     if (printp)
       emitPrintTensor(b, lvar);
-    // dealloc lvar
+  }
+
+  for (auto &vvar : valVars) {
+    b.create<memref::DeallocOp>(loc, vvar);
+  }
+  for (auto &lvar : localVars) {
     b.create<memref::DeallocOp>(loc, lvar);
   }
 
   b.create<func::ReturnOp>(loc, ValueRange{});
 
+  // Wrap the kernels and gather them to substitute in calls.
+  llvm::SmallDenseMap<func::FuncOp, func::FuncOp> wrappedFuncs;
```

Comment:
I could swear I'd already pushed this, but I can't find the pull request.  The commit was from April, though.